apiserver/facades/client/charms/charmhubrepo.go

```diff
@@ -370,14 +371,24 @@ func refreshConfig(curl *charm.URL, origin corecharm.Origin) (charmhub.RefreshCo
 		method = MethodID
 	}
 
+	// origin.Platform.Series could be a series or a version. In reality it will
+	// be a series (focal, groovy), but to be on the safe side we should
+	// validate and fallback if it really isn't a version.
+	// The refresh will fail if it's wrong with a revision not found, which
+	// will be fine for now.
+	series := channelTrack(origin.Platform.Series)
+	version, err := coreseries.VersionSeries(series)
+	if err != nil {
+		version = series
+	}
+
 	var (
 		cfg charmhub.RefreshConfig
-		err error
 
 		platform = charmhub.RefreshPlatform{
 			Architecture: origin.Platform.Architecture,
 			OS:           origin.Platform.OS,
-			Series:       origin.Platform.Series,
+			Series:       version,
```

Comment:
Why are we passing Series here vs Base?
It feels like we are spending a lot of effort to map Platform back to series in order to refresh with series, but Charmhub is passing us back
```
$ curl -XPOST -s https://api.charmhub.io/v2/charms/refresh -H 'Content-type: application/json' -d '{"context": [], "actions": [{"name": "facundo-snappass-test", "base": {"architecture": "amd64", "name": "NA", "channel": "NA"}, "action": "install", "instance-key": "a-test"}]}'  | jq
{
  "error-list": [],
  "results": [
    {
      "charm": null,
      "error": {
        "code": "invalid-charm-platform",
        "extra": {
          "default-bases": [
            {
              "architecture": "amd64",
              "channel": "20.04",
              "name": "ubuntu"
            }
          ],
          "default-platforms": [
            {
              "architecture": "amd64",
              "os": "ubuntu",
              "series": "focal"
            }
          ]
        },
        "message": "Instance key 'a-test' invalid 'platform' in 'install' for charm_id=C3cqgiAPo0liBr4vwGUzTFl297c5SpMs name=facundo-snappass-test"
      },
      "id": "C3cqgiAPo0liBr4vwGUzTFl297c5SpMs",
      "instance-key": "a-test",
      "name": "facundo-snappass-test",
      "released-at": null,
      "result": "error"
    }
  ]
}
```

Which does have platform, but it also has bases, which we can pass back in.

Reply:
This is part of the boundary of where we're using series vs base in juju.  Eventually RefreshPlatform must become RefreshBase, but currently that's hidden inside of charmhub pkg.

Though it brings up a good question of why the change charmhub wasn't enough for this, perhaps that should have been tweaked instead.

Reply:
Perhaps this should have been addressed in here: https://github.com/juju/juju/blob/d25ae7239f01873221fc75cf510b553168c1cdff/charmhub/refresh.go#L443

Reply:
I can fix this tomorrow morning